File: src/my_project_typescript/easy_problems_typescript/from101to150/shortestCompletingWord.ts
function shortestCompletingWord(licensePlate: string, words: string[]): string {

    words = words.reverse()
    words = words.sort( (a,b) => a.length <= b.length ? -1 : 1)

    let stringFiltered = ""
    licensePlate = licensePlate.toLocaleLowerCase()
    let exists: boolean = true

    for (let i = 0; i < licensePlate.length; i++){
        if (licensePlate[i].charCodeAt(0) >= 97 && licensePlate[i].charCodeAt(0) <= 122){
            stringFiltered += licensePlate[i]
        }
    }

   

    function countValues(string1: string, string2: string): number {
        let count:number  = 0
        for (let word of string2){
            if (word == string1){
                count += 1
            }
        }
        return count 
    }

    for (let word of words){
        exists = true 
        for (let char of stringFiltered){
            if(!word.includes(char)){
                exists = false
                break
            }
            else {
                if (countValues(char, word) < countValues(char, stringFiltered)){
                    exists = false 
                    break 
                }
            }

        }
        if (exists){
            return word
        }
    }

    return ""
};

console.log(shortestCompletingWord("1s3 PSt",  ["step","steps","stripe","stepple"]))

console.log(shortestCompletingWord("1s3 456", ["looks","pest","stew","show"]))

console.log(shortestCompletingWord( "Ah71752",  ["suggest","letter","of","husband","easy","education","drug","prevent","writer","old"]))

console.log(shortestCompletingWord( "OgEu755",  ["enough","these","play","wide","wonder","box","arrive","money","tax","thus"]))



let a = 'sd'

let b = 'sdfhendn'

let count = (b.match(/`${a}`/g) || []).length

console.log(count)

File: src/my_project/easy_problems/from201to250/shortest_completing_word.py
from typing import List 
import re 
class Solution:
    def shortestCompletingWord(self, licensePlate: str, words: List[str]) -> str:
        words = sorted(words, key=len)
        string_filterd = ''.join(sorted([x for x in licensePlate.lower() if 97 <= ord(x) and ord(x) <= 122]))
        
        for word in words:
            
            temp = ''.join(word)
            exists = True
            for char in string_filterd:
                if char not in temp: 
                    exists = False
                    break 
                else: 
                    if temp.count(char) < string_filterd.count(char):
                        exists = False 
                        break 
            if exists: 
                return word
        return ""
    
    
soluiton = Solution()

print(soluiton.shortestCompletingWord(licensePlate = "1s3 PSt", words = ["step","steps","stripe","stepple"]))